packages/@react-spectrum/calendar/stories/RangeCalendar.stories.tsx

```diff
@@ -57,12 +58,19 @@ storiesOf('Date and Time/RangeCalendar', module)
     () => render({defaultValue: {start: new CalendarDate(2019, 6, 5), end: new CalendarDate(2019, 6, 10)}, isDisabled: true})
   )
   .add(
-    'multiple disabled intervals',
-    () => render({isDateDisabled: (date: DateValue) => {
+    'isDateUnavailable',
+    () => render({isDateUnavailable: (date: DateValue) => {
       const disabledIntervals = [[today(getLocalTimeZone()), today(getLocalTimeZone()).add({weeks: 1})], [today(getLocalTimeZone()).add({weeks: 2}), today(getLocalTimeZone()).add({weeks: 3})]];
       return disabledIntervals.some((interval) => date.compare(interval[0]) > 0 && date.compare(interval[1]) < 0);
     }})
```

Comment:
I was able to get the following to happen by setting a defaultValue that spans into the disabledInterval:
<img width="366" alt="image" src="https://user-images.githubusercontent.com/9661127/157753847-af246da7-6c7c-4068-944f-0cadbc64fea6.png">
The 12th has the following styles due to it being the end date:
<img width="539" alt="image" src="https://user-images.githubusercontent.com/9661127/157754146-ec5fccca-434a-4acf-b980-97194be52b90.png">
Perhaps it shouldn't get `is-selection-end`, needs some additional logic within CalendarCell?